runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -408,6 +408,17 @@ local function disable(bufnr)
   clear(bufnr)
 end
 
+--- Refresh diagnostics, only if we have attached clients that support it
+---@param bufnr (integer) buffer number
+---@param opts table | nil Additional options to pass to util_refresh
+---@private
+local function _refresh(bufnr, opts)
+  if #vim.lsp.get_clients({ bufnr = bufnr, method = ms.textDocument_diagnostic }) > 0 then
```

Comment:
Can this check be added to `util._refresh` instead? Could be an option on `opts` if there is a case(?) where it makes sense to request things not supported by clients...

Reply:
yeah, good point. I'll move this check into `util._refresh`